templates/backup.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+# Logic is as follows:
+
+# 1. If pre-script fails exit immediately
+# 2. Proceed if backup
+# 3. Irrespective whether backup succeeds execute post-script.
+
+timestamp=$(date -Isecond)
+
+{% if TARSNAP_BACKUP_PRE_SCRIPT %}
+{{ TARSNAP_BACKUP_PRE_SCRIPT }} || exit 1
+{% endif %}
+
+# We run --fsck before backup, so in case of any inconsistencies (that might happen)
+# backups will still work
+
+/usr/local/bin/tarsnap --fsck --keyfile "{{ TARSNAP_KEY_REMOTE_LOCATION }}" \
```

Comment:
@antoviaque Tarsnap assumes that key will be used from only as single machine.  Just to be 100%:  if I disable it, after I delete old backups for machine A, backups on machine A will fail until someone runs fsck manually.